include/rotatingMHD/magnetic_induction.h

```diff
@@ -0,0 +1,490 @@
+ *
+ */
+void correction_step(const bool reinit_preconditioner) override;
+
+/*!
+ * @brief A method assembling the linear algebra of the diffusion step
+ *
+ */
+void assemble_diffusion_step() override;
+
+/*!
+ * @brief A method assembling the right-hand side vector of the
+ * diffusion step
+ *
+ */
+void assemble_diffusion_step_rhs() override;
+
+/*!
+ * @brief This method assembles the local right-hand side of the
+ * diffusion step on a single cell.
+ */
+void assemble_local_diffusion_step_rhs(
+  const typename DoFHandler<dim>::active_cell_iterator             &cell,
+  AssemblyData::MagneticInduction::DiffusionStepRHS::Scratch<dim>  &scratch,
+  AssemblyData::MagneticInduction::DiffusionStepRHS::Copy          &data);
+
+/*!
+ * @brief This method copies the local right-hand side of the diffusion
+ * step into the global vector.
+ */
+void copy_local_to_global_diffusion_step_rhs(
+  const AssemblyData::MagneticInduction::DiffusionStepRHS::Copy  &data);
+
+/*!
+ * @brief A method assembling the right-hand side vector of the
+ * projection step
+ *
+ */
+void assemble_projection_step_rhs() override;
+
+/*!
+ * @brief This method assembles the local right-hand side of the
+ * projection step on a single cell.
+ */
+void assemble_local_projection_step_rhs(
+  const typename DoFHandler<dim>::active_cell_iterator              &cell,
+  AssemblyData::MagneticInduction::ProjectionStepRHS::Scratch<dim>  &scratch,
+  AssemblyData::MagneticInduction::ProjectionStepRHS::Copy          &data);
+
+/*!
+ * @brief This method copies the local right-hand side of the projection
+ * step into the global vector.
+ */
+void copy_local_to_global_projection_step_rhs(
+  const AssemblyData::MagneticInduction::ProjectionStepRHS::Copy  &data);
+
+/*!
+ * @brief A method assembling the right-hand side vector of the
+ * initialization step
+ *
+ */
+void assemble_initialization_step_rhs() override;
+
+/*!
+ * @brief This method assembles the local right-hand side of the
+ * initialization step on a single cell.
+ */
+void assemble_local_initialization_step_rhs(
+  const typename DoFHandler<dim>::active_cell_iterator                  &cell,
+  AssemblyData::MagneticInduction::InitializationStepRHS::Scratch<dim>  &scratch,
+  AssemblyData::MagneticInduction::InitializationStepRHS::Copy          &data);
+
+/*!
+ * @brief This method copies the local right-hand side of the
+ * initialization step into the global vector.
+ */
+void copy_local_to_global_initialization_step_rhs(
+  const AssemblyData::MagneticInduction::InitializationStepRHS::Copy  &data);
+
+/*!
+ * @brief A method assembling advection matrix of the diffusion step
+ *
+ * @details The advection matrix is given by
+ * \f[
+ * \mathcal{A}_{\boldsymbol{\mathfrak{B}}}
+ * =
+ * \int
+ * \bs{q} \cdot [
+ * (\nabla \cdot \boldsymbol{\mathfrak{B}}^{k}) \bs{v}^{k}
+ * +
+ * (\nabla \cdot \bs{v}^{k}) \boldsymbol{\mathfrak{B}}^{k}
+ * +
+ * \boldsymbol{\mathfrak{B}}^{k} \cdot (\nabla \otimes \bs{v}^{k})
```

Comment:
I would interpret this term as a _source term_ and move it to the right-hand side.